File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletRealTimeClock.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletRealTimeClock;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Real-Time Clock Bricklet

$ipcon = new IPConnection(); // Create IP connection
$rtc = new BrickletRealTimeClock(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current date and time
$date_time = $rtc->getDateTime();

echo "Year: " . $date_time['year'] . "\n";
echo "Month: " . $date_time['month'] . "\n";
echo "Day: " . $date_time['day'] . "\n";
echo "Hour: " . $date_time['hour'] . "\n";
echo "Minute: " . $date_time['minute'] . "\n";
echo "Second: " . $date_time['second'] . "\n";
echo "Centisecond: " . $date_time['centisecond'] . "\n";

if ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_MONDAY) {
    echo "Weekday: Monday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_TUESDAY) {
    echo "Weekday: Tuesday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_WEDNESDAY) {
    echo "Weekday: Wednesday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_THURSDAY) {
    echo "Weekday: Thursday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_FRIDAY) {
    echo "Weekday: Friday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_SATURDAY) {
    echo "Weekday: Saturday\n";
} elseif ($date_time['weekday'] == BrickletRealTimeClock::WEEKDAY_SUNDAY) {
    echo "Weekday: Sunday\n";
}

// Get current timestamp
$timestamp = $rtc->getTimestamp();
echo "Timestamp: $timestamp ms\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, real_time_clock_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Real-Time Clock Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let rtc = RealTimeClockBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current date and time.
    let date_time = rtc.get_date_time().recv()?;

    println!("Year: {}", date_time.year);
    println!("Month: {}", date_time.month);
    println!("Day: {}", date_time.day);
    println!("Hour: {}", date_time.hour);
    println!("Minute: {}", date_time.minute);
    println!("Second: {}", date_time.second);
    println!("Centisecond: {}", date_time.centisecond);

    if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_MONDAY {
        println!("Weekday: Monday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_TUESDAY {
        println!("Weekday: Tuesday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_WEDNESDAY {
        println!("Weekday: Wednesday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_THURSDAY {
        println!("Weekday: Thursday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_FRIDAY {
        println!("Weekday: Friday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_SATURDAY {
        println!("Weekday: Saturday");
    } else if date_time.weekday == REAL_TIME_CLOCK_BRICKLET_WEEKDAY_SUNDAY {
        println!("Weekday: Sunday");
    }

    // Get current timestamp.
    let timestamp = rtc.get_timestamp().recv()?;
    println!("Timestamp: {} ms", timestamp);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
